test/basics.spec.js

```diff
@@ -532,4 +532,145 @@ describe('MockAdapter basics', function() {
+      })
+      .then(function(response) {
+        expect(response.data.id).to.equal(3);
+        return instance.get('/users', { params: { searchText: 'Jackie' } });
+      })
+      .then(function(response) {
+        expect(response.data.id).to.equal(4);
+      });
+  });
+
+  it('should overwrite replys', function() {
+    mock.onGet('/').reply(500);
+    mock.onGet('/').reply(200);
+    mock.onGet('/').reply(401);
+
+    return instance.get('/')
+      .catch(function(error) {
+        expect(mock.handlers['get'].length).to.equal(1);
+        expect(error.response.status).to.equal(401);
+      });
+  });
+
+  it.only('should overwrite replys using RegEx', function() {
+    mock.onGet(/foo\/bar/).reply(500);
+    mock.onGet(/foo\/bar/).reply(200);
+    mock.onGet(/foo\/baz\/.+/).reply(200);
+
+    return instance.get('/foo/bar')
+      .then(function(response) {
+        expect(mock.handlers['get'].length).to.equal(2);
+        expect(response.status).to.equal(200);
+        return instance.get('/foo/baz/56');
+      })
+      .then(function(response) {
+        expect(response.status).to.equal(200);
+      });
+  });
+
+  it('should allow overwriting only on reply if replyOnce was used first', function() {
+    var counter = 0;
+    mock.onGet('/').replyOnce(500);
+    mock.onGet('/').reply(200);
+    mock.onGet('/').reply(401);
+
+    return instance.get('/')
+      .catch(function(error) {
+        expect(error.response.status).to.equal(500);
+        counter += 1;
+        return instance.get('/');
+      })
+      .catch(function(error) {
+        expect(error.response.status).to.equal(401);
+        counter += 1;
+      })
+      .then(function() {
+        expect(counter).to.equal(2);
+      });
+  });
+
+  it('should not allow overwriting only on reply if replyOnce wasn\'t used first', function() {
+    var counter = 0;
+    mock.onGet('/').reply(200);
+    mock.onGet('/').reply(401);
+    mock.onGet('/').replyOnce(500);
+    mock.onGet('/').reply(500);
+
+    return instance.get('/')
+      .catch(function(error) {
+        expect(error.response.status).to.equal(500);
+        counter += 1;
+        return instance.get('/');
+      })
+      .catch(function(error) {
+        expect(error.response.status).to.equal(500);
+        counter += 1;
+      })
+      .then(function() {
+        expect(counter).to.equal(2);
+      });
+  });
+  it('allows overwriting mocks with parameters', function() {
+    mock
+      .onGet('/users', { params: { searchText: 'John' } })
+      .reply(500)
+      .onGet('/users', { params: { searchText: 'John' } })
+      .reply(200, { id: 1 });
+
+    return instance.get('/users', { params: { searchText: 'John' } })
+      .then(function(response) {
+        expect(response.status).to.equal(200);
+      });
+  });
+
+  it('allows overwriting mocks with headers', function() {
+    mock.onGet('/').reply(200, {}, { test: true });
+    mock.onGet('/').reply(200, {}, { test: false });
```

Comment:
This isn't testing what it intends to test, the headers should be part of the call to `onGet` to match requests according to the request headers.

Reply:
Addressed